pkg/aws/eni/node.go

```diff
@@ -289,6 +292,8 @@ func indexExists(enis map[string]eniTypes.ENI, index int64) bool {
 // findNextIndex returns the next available index with the provided index being
 // the first candidate
 func (n *Node) findNextIndex(index int64) int64 {
```

Comment:
It's odd seeing a private method getting a mutext. Please add a comment in the function stating `n.mutex` must NOT be held.